packages/core/src/shared/extensions/ssh.ts

```diff
@@ -119,6 +123,30 @@ export class RemoteSshSettings extends Settings.define('remote.SSH', remoteSshTy
     }
 }
 
+export async function testSshConnection(
+    ProcessClass: typeof ChildProcess,
+    hostname: string,
+    sshPath: string,
+    user: string,
+    session: SSM.StartSessionResponse
+): Promise<ChildProcessResult | never> {
+    try {
+        const env = { SESSION_ID: session.SessionId, STREAM_URL: session.StreamUrl, TOKEN: session.TokenValue }
+        const result = await new ProcessClass(sshPath, [
```

Comment:
Because we are running it in a bound process, this becomes a little tricky. We can pass a process as an argument, but that opens the door for processes that don't behave how `tryRun` is expecting. For example, in this case our bound process rejects on errors. `tryRun` does not expect this. `tryRun` would need to be refactored to manually handle all `spawnOptions` that could alter behavior and these might be specific to the use case.